src/main.ts

```diff
@@ -214,6 +214,21 @@ export class Purchases {
           "Creating and returning new instance.",
       );
     }
+
+    if (!apiKey) {
+      throw new PurchasesError(
+        ErrorCode.ConfigurationError,
+        "API key is required. Please provide a valid RevenueCat API key.",
+      );
+    }
+
+    if (!appUserId) {
+      throw new PurchasesError(
+        ErrorCode.ConfigurationError,
+        "App user ID is required. Please provide a valid user identifier.",
+      );
+    }
```

Comment:
Hmm shouldn’t the validators a couple lines below be catching these? https://github.com/RevenueCat/purchases-js/blob/main/src/helpers/configuration-validators.ts

Though looking at it, maybe an undefined value could go through the app user id validator… In that case maybe it’s worth moving this check there?

Reply:
sorry still not ready for review, should I have put it up as a draft, but makes sense